pywhy_graphs/algorithms/generic.py

```diff
@@ -604,3 +605,85 @@ def inducing_path(G, node_x: Node, node_y: Node, L: Set = None, S: Set = None):
                 break
 
     return (path_exists, path)
+
+
+def _find_adc(G):
```

Comment:
This can be a public function and tested separately to help you ensure that this works as intended.

Reply:
Sorry for the late response, you're right. I'll do just that.